conftest.py

```diff
@@ -221,6 +224,37 @@ def pytest_terminal_summary(
         terminalreporter.write("\n" * 2)
 
 
+_MLFLOW_VERSION_NOT_FOUND_ERROR_REGEX = re.compile(
+    r"No matching distribution found for mlflow==([0-9]+\.[0-9]+\.[0-9]+(\.dev[0-9]+)?)"
+)
+
+
+def pytest_exception_interact(node, call, report):
```

Comment:
Curious what pytest output looks like.

Reply:
Looks like this;
```
tests/models/test_utils.py::test_adding_libraries_to_model_default FAILED | MEM 0.9/61.8 GB | DISK 71.7/484.6 GB [ 20%]
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/_pytest/main.py", line 271, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/_pytest/main.py", line 325, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/_pytest/main.py", line 350, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/_pytest/runner.py", line 114, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/_pytest/runner.py", line 133, in runtestprotocol
INTERNALERROR>     reports.append(call_and_report(item, "call", log))
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/_pytest/runner.py", line 228, in call_and_report
INTERNALERROR>     hook.pytest_exception_interact(node=item, call=call, report=report)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_callers.py", line 113, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/.venv/cloudpickle2/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/conftest.py", line 247, in pytest_exception_interact
INTERNALERROR>     raise MlflowException(
INTERNALERROR> mlflow.exceptions.MlflowException: The version of MLflow you're attempting to install (2.8.1.dev0) is not found on PyPI. This could be due to specifying a local development version like 2.8.1.dev0, which is one micro version ahead of the latest official release. To resolve this issue, you can use the `--serve-wheel` flag with your pytest command, for example: `pytest tests --serve-wheel`. This flag sets up a local PyPI server, allowing you to install MLflow from the local source code.
```

Just realized that it hides original traceback, maybe we should show it

Reply:
Hmm, `INTERNALERROR>` looks a bit confusing and might hide what went wrong.

Reply:
by doing `raise MlflowException(...).with_traceback(exception.__traceback__)`, it will be like

```
...
INTERNALERROR>     return Model.log(
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/mlflow/models/model.py", line 619, in log
INTERNALERROR>     flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/mlflow/models/wheeled_model.py", line 128, in save_model
INTERNALERROR>     WheeledModel._download_wheels(
INTERNALERROR>   File "/home/yuki.watanabe/mlflow/mlflow/models/wheeled_model.py", line 223, in _download_wheels
INTERNALERROR>     raise MlflowException(
INTERNALERROR> mlflow.exceptions.MlflowException: The version of MLflow you're attempting to install (2.8.1.dev0) is not found on PyPI. This could be due to specifying a local development version like 2.8.1.dev0, which is one micro version ahead of the latest official release. To resolve this issue, you can use the `--serve-wheel` flag with your pytest command, for example: `pytest tests --serve-wheel`. This flag sets up a local PyPI server, allowing you to install MLflow from the local source code.
```
it seems we can't replace `INTERNALERROR` part tho

Reply:
How about this?
![Screenshot 2023-11-02 at 15 08 03](https://github.com/mlflow/mlflow/assets/31463517/ff247872-e352-4ecb-b825-bcacc10a6485)

Reply:
I think `INTERNALERROR>` still looks scary.


@serena-ruan @daniellok-db what do you think?

Reply:
IT IS scary 🤣 Can we filter out the INTERNALERROR prefix?

Reply:
It does look scary but if it's hard to remove I think it's ok. I usually just read the last error message anyway haha

Reply:
Thanks both! Unfortunately we can't remove it... as long as we throw an exception from the hook.
I do not like it too....but probably ok given that at least only developers see that, i.e. we don't show this to normal user and let them like "omg".

Reply:
what do you think about this approach? https://github.com/mlflow/mlflow/pull/10264#issuecomment-1790000666